### PR TITLE
V14 QA Added integration tests to handle authorization of DataType

### PR DESCRIPTION
**Pull request by nhudinh0309**
### Description
Added integration tests about authorization handlers for DataType
Removed unused class and methods corresponding to old Base Class solution

### CODE REVIEW

**Comment by JasonElkin** [comment]
Core collabs can ignore this.

**Comment by JasonElkin** [comment]
Core collabs can ignore this.

**Comment by andr317c** [comment]
Needs refactoring to match the new best practice :D

File: tests/Umbraco.Tests.Integration/ManagementApi/DataType/DeleteDataTypeControllerTests.cs [hunk 1]
```diff
@@ -1,80 +1,46 @@
...
     {
-        var response = await Client.DeleteAsync(Url);
+        ExpectedStatusCode = HttpStatusCode.Unauthorized
+    };
 
-        Assert.AreEqual(HttpStatusCode.Unauthorized, response.StatusCode, await response.Content.ReadAsStringAsync());
+    protected override async Task<HttpResponseMessage> ClientRequest()
+    {
+        return await Client.DeleteAsync(Url);
     }
```
Comment (andr317c) [inline tests/Umbraco.Tests.Integration/ManagementApi/DataType/DeleteDataTypeControllerTests.cs:45]:
```suggestion
    protected override async Task<HttpResponseMessage> ClientRequest() => await Client.DeleteAsync(Url);
```

**Review by andr317c — CHANGES_REQUESTED** [review]

**Review by andr317c — APPROVED** [review]

**Pull request merged**